funnel/forms/proposal.py

```diff
@@ -74,10 +74,15 @@ def __init__(self, *args, **kwargs):
             self.description.description = space.proposal_part_b.get('hint')
 
 
-class ProposalStatusForm(forms.Form):
-    status = forms.SelectField(
-        __("Status"), coerce=int,
-        choices=[(status, label.title) for (status, label) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT])
+class ProposalTransitionForm(forms.Form):
+    transition = forms.SelectField(__("Status"), validators=[forms.validators.DataRequired()])
```

Comment:
Add a validation to check if the supplied choice is a valid transition.

Reply:
The `getattr` call below will fail if it isn't.

Reply:
@shreyas-satish the validation is done with the choices. It's already happening because I'm filling up the choices.

Reply:
That's assuming a user submits the form through the user interface. In this context this will likely not be a problem, but generally it's safer to not trust user input.

Reply:
> It's already happening because I'm filling up the choices.

That isn't a validation. That's just pre-populating an input element with choices. A user can still supply a value through JavaScript that isn't listed in the choices.